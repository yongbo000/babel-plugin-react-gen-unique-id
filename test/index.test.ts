import pluginTester from 'babel-plugin-tester'
import path from 'path'
import plugin from '../src'

pluginTester({
  plugin,
  babelOptions: { parserOpts: { plugins: ['jsx'] } },
  snapshot: true,
  pluginName: 'babel-plugin-react-auto-unique-id',
  tests: [
    {
      title: 'Class Component',
      fixture: path.join(__dirname, '__fixtures__', 'classComponent.tsx'),
    },
    {
      title: 'Functional Component',
      fixture: path.join(__dirname, '__fixtures__', 'functionComponent.tsx'),
    },
    {
      title: 'Arrow Function Component',
      fixture: path.join(__dirname, '__fixtures__', 'arrowFunction.tsx'),
    },

    {
      title: 'Property exists',
      fixture: path.join(__dirname, '__fixtures__', 'propertyExists.tsx'),
    },
    {
      title: 'Same Siblings',
      fixture: path.join(__dirname, '__fixtures__', 'sameSiblings.tsx'),
    },
  ],
})
