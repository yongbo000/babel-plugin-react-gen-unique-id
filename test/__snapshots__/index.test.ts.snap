// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-react-auto-unique-id Arrow Function Component: Arrow Function Component 1`] = `

import ComA from './components/ComA';

const Div2 = () => <div />

const DivBody = () => {
  return <div>
    <p>hello</p>
    <ComA />
  </div>
}

      ↓ ↓ ↓ ↓ ↓ ↓

import ComA from './components/ComA'
const Div2 = () => <div data-spm="_test___fixtures___arrowFunction_Div2_div" />
const DivBody = () => {
  return (
    <div data-spm="_test___fixtures___arrowFunction_DivBody_div">
      <p data-spm="_test___fixtures___arrowFunction_DivBody_p">hello</p>
      <ComA data-spm="_test___fixtures___arrowFunction_DivBody_ComA" />
    </div>
  )
}


`;

exports[`babel-plugin-react-auto-unique-id Class Component: Class Component 1`] = `

class ComA extends React.Component {
  render() {
    return <div>CCCCC</div>
  }
}

const DivBody = () => {
  return <div>
    <ComA />
    <div>hello</div>
  </div>
}

      ↓ ↓ ↓ ↓ ↓ ↓

class ComA extends React.Component {
  render() {
    return (
      <div data-spm="_test___fixtures___classComponent_ComA_div">CCCCC</div>
    )
  }
}
const DivBody = () => {
  return (
    <div data-spm="_test___fixtures___classComponent_DivBody_div">
      <ComA data-spm="_test___fixtures___classComponent_DivBody_ComA" />
      <div data-spm="_test___fixtures___classComponent_DivBody_div_1">
        hello
      </div>
    </div>
  )
}


`;

exports[`babel-plugin-react-auto-unique-id Component: Component 1`] = `

class ComB extends React.Component {
  render() {
    return <div>
      <p>BBB</p>
    </div>
  }
}

const ComA = () => {
  return (
    <div className="ComA">
      <>
      i'am A
      </>
      {[1,2,3].map(() => {
        return <ComB />;
      })}
      {[1,2].map(() => (<div>1111</div>))}
    </div>
  );
}

export default ComA;

      ↓ ↓ ↓ ↓ ↓ ↓

class ComB extends React.Component {
  render() {
    return (
      <div data-spm="_test___fixtures___components_ComA_index_ComB_div">
        <p data-spm="_test___fixtures___components_ComA_index_ComB_p">BBB</p>
      </div>
    )
  }
}
const ComA = () => {
  return (
    <div
      className="ComA"
      data-spm="_test___fixtures___components_ComA_index_ComA_div"
    >
      <>i'am A</>
      {[1, 2, 3].map(() => {
        return (
          <ComB data-spm="_test___fixtures___components_ComA_index_ComA_ComB" />
        )
      })}
      {[1, 2].map(() => (
        <div data-spm="_test___fixtures___components_ComA_index_ComA_div_1">
          1111
        </div>
      ))}
    </div>
  )
}
export default ComA


`;

exports[`babel-plugin-react-auto-unique-id Functional Component: Functional Component 1`] = `

function Div() {
  return <div />
}

function Nested() {
  return (
    <div>
      hello
      <A />
      <div>world</div>
    </div>
  )
}

      ↓ ↓ ↓ ↓ ↓ ↓

function Div() {
  return <div data-spm="_test___fixtures___functionComponent_Div_div" />
}
function Nested() {
  return (
    <div data-spm="_test___fixtures___functionComponent_Nested_div">
      hello
      <A data-spm="_test___fixtures___functionComponent_Nested_A" />
      <div data-spm="_test___fixtures___functionComponent_Nested_div_1">
        world
      </div>
    </div>
  )
}


`;

exports[`babel-plugin-react-auto-unique-id Property exists: Property exists 1`] = `

function Sibling() {
  return <div />
}

function Siblings() {
  return (
    <div>
      hello
      <Sibling data-id="i-exist"/>
      <Sibling/>
    </div>
  )
}

      ↓ ↓ ↓ ↓ ↓ ↓

function Sibling() {
  return <div data-spm="_test___fixtures___propertyExists_Sibling_div" />
}
function Siblings() {
  return (
    <div data-spm="_test___fixtures___propertyExists_Siblings_div">
      hello
      <Sibling
        data-id="i-exist"
        data-spm="_test___fixtures___propertyExists_Siblings_Sibling"
      />
      <Sibling data-spm="_test___fixtures___propertyExists_Siblings_Sibling_1" />
    </div>
  )
}


`;

exports[`babel-plugin-react-auto-unique-id Same Siblings: Same Siblings 1`] = `

function Sibling() {
  return <div />
}

function Siblings() {
  return (
    <div>
      hello
      <Sibling/>
      <Sibling/>
    </div>
  )
}

      ↓ ↓ ↓ ↓ ↓ ↓

function Sibling() {
  return <div data-spm="_test___fixtures___sameSiblings_Sibling_div" />
}
function Siblings() {
  return (
    <div data-spm="_test___fixtures___sameSiblings_Siblings_div">
      hello
      <Sibling data-spm="_test___fixtures___sameSiblings_Siblings_Sibling" />
      <Sibling data-spm="_test___fixtures___sameSiblings_Siblings_Sibling_1" />
    </div>
  )
}


`;
